src/DataWrangling.jl

```diff
@@ -0,0 +1,57 @@
+module DataWrangling
+
+export continue_downwards!
+
+using Oceananigans.Grids: peripheral_node
+using Oceananigans.Utils: launch!
+using Oceananigans.Fields: instantiated_location, interior
+using Oceananigans.Architectures: architecture, device_event, device, GPU
+
+using KernelAbstractions: @kernel, @index
+using KernelAbstractions.Extras.LoopInfo: @unroll
+
+using ImageInpainting
+
+function continue_downards!(field)
+    arch = architecture(field)
+    grid = field.grid
+    loc = instantiated_location(field)
+
+    event = launch!(arch, grid, :xy, _continue_downwards!, field, loc, grid; dependencies=device_event(arch))
+
+    wait(device(arch), event)
+
+    return nothing
+end
+
+@kernel function _continue_downwards!(field, (LX, LY, LZ), grid)
+    i, j = @index(Global, NTuple)
+
+    Nz = grid.Nz
+    active_surface = !peripheral_node(i, j, Nz, grid, LX, LY, LZ)
+
+    @unroll for k = Nz-1 : -1 : 1
+        fill_from_above = active_surface & peripheral_node(i, j, k, grid, LX, LY, LZ)
+        @inbounds field[i, j, k] = ifelse(fill_from_above, field[i, j, k+1], field[i, j, k])
+    end
+end
+
+function inpaint_horizontally!(field; algorithm=Criminisi(5, 5))
+    arch = architecture(field)
+    grid = field.grid
+    loc = instantiated_location(field)
+    Nx, Ny, Nz = size(grid)
+
+    # Could transfer to CPU, then transfer back.
+    arch isa GPU && error("Inpainting on the GPU is not supported yet!")
+
+    for k = 1:Nz
+        mask = [peripheral_node(i, j, k, grid, loc...) for i = 1:Nx, j = 1:Ny]
```

Comment:
```suggestion
        mask = [peripheral_node(i, j, k, grid, loc...) for j = 1:Ny, i = 1:Nx]
```

Reply:
huh, it seemed to be working

Reply:
isn't `i` the first index, `j` the second?

Reply:
yes but performance-wise is better to loop over columns first, right?

Reply:
https://docs.julialang.org/en/v1/manual/performance-tips/#man-performance-column-major

Reply:
That's up to the generator, not me

Reply:
there is a good chance I got confused here

Reply:
I'm also wasting a lot of memory, it would be more efficient to write

```julia
mask = zeros(Nx, Ny)

for k = 1:Nz
    for i = 1:Nx
        for j = 1:Ny
            mask[i, j] = peripheral_node(i, j, k, grid, loc...)
        end
    end
end
```

Then we have column major ordering for the loop as you suggest and no allocation.

However, note that the generator:

```julia
mask = [peripheral_node(i, j, k, grid, loc...) for j = 1:Ny, i = 1:Nx]
```

may create the array in whatever order is most efficient.

Reply:
I think though that the cost here is completely dominated by the inpainting...

Reply:
I will check to be sure.

Reply:
okie!
either way this step only happens once so it's not at all an issue!